sdk/nodejs/apigateway/experimental/api.ts

```diff
@@ -351,19 +400,29 @@ function addRoutesToSwaggerSpec(
     }
 }
 
-function addStaticRoutesToSwaggerSpec(
-    name: string, swagger: SwaggerSpec, routes: StaticRoute[], opts: pulumi.ResourceOptions) {
-
-    // If there are no static files or directories, then we can bail out early.
-    if (!routes || routes.length === 0) {
-        return swagger;
-    }
+function addStaticRouteToSwaggerSpec(
+    name: string, swagger: SwaggerSpec, path: string, route: StaticRoute, opts: pulumi.ResourceOptions) {
 
     const method = swaggerMethod("GET");
 
     // Create a bucket to place all the static data under.
     const bucket = new aws.s3.Bucket(safeS3BucketName(name), undefined, opts);
 
+    // For each static file, just make a simple bucket object to hold it, and create a swagger path
+    // that routes from the file path to the arn for the bucket object.
+    //
+    // For static directories, use greedy api-gateway path matching so that we can map a single api
+    // gateway route to all the s3 bucket objects we create for the files in these directories.
+    const stat = fs.statSync(route.localPath);
+    if (stat.isFile()) {
+        processFile(route);
+    }
+    else if (stat.isDirectory()) {
+        processDirectory(route);
+    }
+
+    return;
```

Comment:
same logic as before, just withotu the foreach loop over the routes since that is handled by our caller.